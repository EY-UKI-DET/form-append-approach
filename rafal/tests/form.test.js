/**
 * @jest-environment jsdom
 */

import {
  getByLabelText,
  getByText,
  getByTestId,
  queryByTestId,
  waitFor,
} from "@testing-library/dom";
// adds special assertions like toHaveTextContent
import "@testing-library/jest-dom";

function getExampleDOM() {
  const form = document.createElement("form");
  form.innerHTML = `
      <div class="input-container ic1">
          <label for="name">Name:</label>
          <input id="name" class="input" type="text" required aria-label="Name" />
      </div>
      <button type="submit" class="submit">Submit</button>
  `;

  const button = form.querySelector("button");
  const input = form.querySelector("input");

  button.addEventListener("click", () => {
    setTimeout(() => {
      const printedNameContainer = document.createElement("div");
      printedNameContainer.innerHTML = `
        <div data-testid="printed-name">${input.value}</div>
      `;
      form.appendChild(printedNameContainer);
    }, Math.floor(Math.random() * 200));
  });

  return form;
}

test("examples of some things", async () => {
  const famousProgrammerInHistory = "Rafal";
  const container = getExampleDOM();

  const input = getByLabelText(container, "Name");
  input.value = famousProgrammerInHistory;

  getByText(container, "Submit").click();

  await waitFor(() =>
    expect(queryByTestId(container, "printed-name")).toBeTruthy()
  );

  expect(getByTestId(container, "printed-name")).toHaveTextContent(
    famousProgrammerInHistory
  );

  expect(container).toMatchSnapshot();
});
