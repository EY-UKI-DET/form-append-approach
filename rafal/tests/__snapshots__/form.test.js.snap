// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples of some things 1`] = `
<form>
  
      
  <div
    class="input-container ic1"
  >
    
          
    <label
      for="name"
    >
      Name:
    </label>
    
          
    <input
      aria-label="Name"
      class="input"
      id="name"
      required=""
      type="text"
    />
    
      
  </div>
  
      
  <button
    class="submit"
    type="submit"
  >
    Submit
  </button>
  
  
  <div>
    
        
    <div
      data-testid="printed-name"
    >
      Rafal
    </div>
    
      
  </div>
</form>
`;
